### PR TITLE
feat: (experimental) make error suggestion pluggable

### PR DESCRIPTION
**Pull request by czgu**
You can now insert a custom error message whenever there is a query error
image

### CODE REVIEW

File: querybook/webapp/lib/query-result/error-suggestion.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+                message: string;
+            };
+        };
+    }
+);
+
+function getDefaultQueryErrorSuggestion(
+    queryError: IQueryError,
+    _queryExecution: IQueryExecution,
+    _statementExecutions: IStatementExecution[],
```
Comment (jczhong84) [inline querybook/webapp/lib/query-result/error-suggestion.ts:61]: seems these two parameters are not used, why need them?

File: querybook/webapp/lib/query-result/error-suggestion.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            statementExecutions,
+            queryEngine
+        );
+    }
+    return getDefaultQueryErrorSuggestion(
+        queryError,
+        queryExecution,
+        statementExecutions,
+        queryEngine
+    );
```
Comment (jczhong84) [inline querybook/webapp/lib/query-result/error-suggestion.ts:96]: nit: can simplify it as 
```
const getQueryErrorSuggestion = window.GET_QUERY_ERROR_SUGGESTION ?? getDefaultQueryErrorSuggestion;
return getQueryErrorSuggestion(
        queryError,
        queryExecution,
        statementExecutions,
        queryEngine
    );
```

File: docs_website/docs/integrations/customize_html.md [hunk 3]
```diff
@@ -191,3 +191,28 @@ window.DEFAULT_ROW_LIMIT = 4096;
 // this removes the option for user to pick limit: none
 window.ALLOW_UNLIMITED_QUERY = false;
 ```
+
+### (Experimental) Custom error tooltip
+
+Whenever there is a query error, you can provide a custom suggestion message by providing this function:
+
+```js
+window.GET_QUERY_ERROR_SUGGESTION?: (
```
[the remark this answers is not in the document]
Comment (czgu) [reply docs_website/docs/integrations/customize_html.md:200]: yeah its hard to connect js files together, unlike python :(

File: querybook/webapp/lib/query-result/error-suggestion.ts (hunk 1 above)
Comment (czgu) [reply querybook/webapp/lib/query-result/error-suggestion.ts:61]: i did the nit part so I guess these parameters are required now 😂

**Review by jczhong84 — APPROVED** [review]

**Pull request merged**